dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeClient.java

```diff
@@ -210,6 +215,38 @@ private long calculateLeastDuration(int time) {
         }
     }
 
+    private boolean shouldHeartbeat(URL url) {
+        String transporter = url.getParameter(CLIENT_KEY, url.getParameter(TRANSPORTER_KEY, "netty"));
+        return !transporter.equalsIgnoreCase("netty") && !transporter.equalsIgnoreCase("netty4");
+    }
+
+    private boolean shouldReconnect(URL url) {
+        boolean reconnect = url.getParameter(Constants.SEND_RECONNECT_KEY, false);
+        if (!reconnect) {
+            String param = url.getParameter(Constants.RECONNECT_KEY, "true");
+            if ("false".equalsIgnoreCase(param)) {

```

Comment:
`RECONNECT_KEY`
Is RECONNECT_KEY a boolean type or string? I am a little confused.
https://github.com/apache/incubator-dubbo/pull/3341/files#diff-db21f00081f086cb06625a6fad1c2e6aR486

Reply:
Seems like it's designed to accept both boolean an integer.

Reply:
yes, it's correct. see my comments https://github.com/apache/incubator-dubbo/pull/3341#issuecomment-457987238